RailwayCo/Assets/Scripts/GameLogic/GameLogic.cs

```diff
@@ -109,6 +109,22 @@ public DepartStatus OnTrainDeparture(Guid train)
         StationMaster.RemoveTrainFromStation(sourceStation, train);
         return DepartStatus.Success;
     }
+    public void OnTrainRemoval(Guid train)
+    {
+        TrainMaster.CompleteTravelPlan(train);
+        TrainMaster.DeactivateTrain(train);
```

Comment:
There is no need to deactivate; the train GameObject from the front end will be destroyed during a collision (intended effect) and the next new train will be a fresh train with a new GUID

On that note, there will be 2 potential situations (down the road)
1. 2 trains collide. Train are destroyed completely (can keep the mtd name OnTrainRemoval)
2. user place them in the train storage, then yea this function will be useful (maybe use a new method called StoreTrain)

Reply:
Oh I see. I will rename this method to OnTrainCollision then. And have a separate method for OnTrainRemoval.
Actly added train active/inactive status to address #156 so yea.

Reply:
Fixed in 8bd03b4 and e108c24